### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by kocsislaci**

### CODE REVIEW

File: kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.github.paylike.kotlin_client.dto
+
+// Describes the hints array received after executing a challenge successfully.
+class HintsResponse(json: Map<String, Any>){
```
Comment (Roverr) [inline kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt]: HintsResponse is
```
{
  "hints": ["first", "second"]
}
```
In which the array is not guaranteed to have multiple members, but guaranteed to have at least one

File: kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    val hints: List<String> = listOf()
+    init {
+        (json["hints"] as List<Any>).forEach { hints.plus(it.toString()) }
```
Comment (Roverr) [inline kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt]: plus returns a list but does not extend your list

`listOf` is immutable
use `mutableListOf` and with that you can use `add()` which will extend your list!

File: kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt (hunk 1 above)
Comment (Roverr) [inline kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt]: Also `json` is going to be `JsonObject` in your case as `kotlinx.serialization` works with that class

File: kotlin_client/src/main/java/com/github/paylike/kotlin_client/request/PaylikeRequestBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.github.paylike.kotlin_client.request
+
+import java.util.concurrent.Future
+
+// PaylikeRequestBuilder provides a flexible way to add
```
Comment (Roverr) [inline kotlin_client/src/main/java/com/github/paylike/kotlin_client/request/PaylikeRequestBuilder.kt]: Are these comments picked up by the Intellisense? 

I've been using [KDoc](https://kotlinlang.org/docs/kotlin-doc.html#kdoc-syntax) assuming that's the only one getting picked up 😄

File: kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt (hunk 1 above)
Comment (Roverr) [reply kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt]: This is true for all of the DTOs pretty much, every map (e.g. `{ "foo": "bar" }` will be `JsonObject`, every list will become `JsonArray` and every value is `JsonPrimitiveValue` if I remember correctly 😄 But I think this is not the de facto fastest way to do this, so I assume you could also use annotations or something to generate auto deserializers for classes. Despite that, I prefer it this way so we can explicitly see what is going on.

**Comment by kocsislaci** [comment]
Now the initial implementation is finished. Needs review and probably a bunch of fixes... Now starting the example

**Pull request merged**